graphein/datasets/pdb_data.py

```diff
@@ -0,0 +1,1092 @@
+        """
+        Select molecules shorter than a given length.
+
+        :param length: Maximum length of molecule.
+        :type length: int
+        :param update: Whether to modify the DataFrame in place, defaults to
+            ``False``.
+        :type update: bool, optional
+
+        :return: DataFrame of selected molecules.
+        :rtype: pd.DataFrame
+        """
+        df = self.df.loc[self.df.length < length]
+
+        if update:
+            self.df = df
+        return df
+
+    def resolution_higher_than_or_equal_to(
+        self, resolution: int, update: bool = False
+    ) -> pd.DataFrame:
+        """Select molecules with a resolution higher than or equal to the given value.
+
+        Conventions for PDB resolution values are used, where a lower resolution
+        value indicates a higher resolution for a molecule overall.
+
+        :param resolution: Minimum resolution of molecule.
+        :type resolution: int
+        :param update: Whether to modify the DataFrame in place, defaults to
+            ``False``.
+        :type update: bool, optional
+
+        :return: DataFrame of selected molecules.
+        :rtype: pd.DataFrame
+        """
+        df = self.df.loc[self.df.resolution <= resolution]
+
+        if update:
+            self.df = df
+        return df
+
+    def resolution_lower_than_or_equal_to(
+        self, resolution: int, update: bool = False
+    ) -> pd.DataFrame:
+        """Select molecules with a resolution lower than or equal to the given value.
+
+        Conventions for PDB resolution values are used, where a higher resolution
+        value indicates a lower resolution for a molecule overall.
+
+        :param resolution: Maximum resolution of molecule.
+        :type resolution: int
+        :param update: Whether to modify the DataFrame in place, defaults to
+            ``False``.
+        :type update: bool, optional
+
+        :return: DataFrame of selected molecules.
+        :rtype: pd.DataFrame
+        """
+        df = self.df.loc[self.df.resolution >= resolution]
+
+        if update:
+            self.df = df
+        return df
+
+    def oligomeric(self, oligomer: int = 1, update: bool = False):
+        """
+        Select molecules with a given oligmeric length.
+
+        :param length: Oligomeric length of molecule, defaults to ``1``.
+        :type length: int
+        :param update: Whether to modify the DataFrame in place, defaults to
+            ``False``.
+        :type update: bool, optional
+
+        :return: DataFrame of selected oligmers.
+        :rtype: pd.DataFrame
+        """
+        df = self.df.loc[self.df.length == oligomer]
+
+        if update:
+            self.df = df
+        return df
+
+    def has_ligand(self, ligand: str, update: bool = False) -> pd.DataFrame:
+        """
+        Select molecules that contain a given ligand.
+
+        :param ligand: Ligand to select. (PDB ligand code)
```

Comment:
We should direct users to relevant PDB docs for ligand codes.

Reply:
Addressed in [this commit](https://github.com/a-r-j/graphein/pull/272/commits/3b42dab1be3429af2e9b2c5da59166bc722c8046).